forms/upload.go

```diff
@@ -0,0 +1,105 @@
+package forms
+
+import (
+	"mime/multipart"
+
+	"go-google-scraper-challenge/helpers"
+	"go-google-scraper-challenge/models"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/beego/beego/v2/core/validation"
+)
+
+type UploadForm struct {
+	File       multipart.File
+	FileHeader *multipart.FileHeader
+	User       *models.User          `valid:"Required"`
+	keywords   []string
+}
+
+// Valid adds custom validation to upload form, sets error when the validation failed.
+func (uf *UploadForm) Valid(v *validation.Validation) {
+	if !uf.validateFilePresence() {
```

Comment:
Setting error and log inside the `validateFilePresence` would be more cleaner I think. This is optional also feedback.
```Go	
err := v.SetError("File", "File cannot be empty")
if err == nil {
  logs.Info("Failed to set error on validation")
}
```

Reply:
@junan Yes, I agreed. But to set the error within the functions we need to pass the `validation` to each function 🤔  let me find the better way to do it then I'll refactor all form validations later on [another chore task](https://github.com/carryall/go-google-scraper-challenge/issues/60)